rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,143 @@
+
+![CARBON](./../../.assets/button_design_system.png)
+
+and the `All Buttons` story for testing purposes
+
+![CARBON](./../../.assets/all_buttons_story.png)
+
+Existing functional or event tests which Cypress is required for will need to be amended to test in the `Docs` tab of the new refactored stories rather than in the `Canvas`. The accessibility tests should not be affected. Any visual verification tests currently done by Cypress alone should be done by Applitools. Any additional cases, such as edge cases, cases that cannot be tested in the component in the `Design System` directory or extended test cases e.g. tests for special characters in labels, should be added in an additional story in the `Test` directory. If all of the configurations can be demonstrated in a single story in the `Test` directory then they should, with visual regression testing run against those stories. We should, however, take care to test only what we deem as bringing value to our regression testing and not have Applitools snapshot everything just because we can.
+
+## Test stories
+The visual test stories in Storybook should be minimal with as many configurations of the components as is sensibly possible presented in one frame to reduce the number of snapshots taken by Applitools. Applitools has the ability to knit together snapshots taken of a scrolling page so having a large number of examples for some components in one frame should not pose a problem. An example of a similar method being used for regression testing elsewhere in the industry is in the Material-UI component library which uses Argos rather than Applitools, some results of which can be found here: https://www.argos-ci.com/mui-org/material-ui/builds/294. Taking their Button component snapshot result as an example, their various size configs for primary, secondary, tertiary and icon button types are presented in one shot:
+
+![CARBON](./../../.assets/material-ui-argos-buttons-example.png)
+
+In this example alone, 13 test cases in our current regression suite could be reduced to just 1 snapshot from Applitools.
+
+## Test feature files
+Cypress will use the feature files to navigate to each story, as it does currently, but without setting all the configuration via the knobs. A step will be added for Applitools to snapshot the canvas. This will remove the need for the numerous steps taken just to get the component into the pre-requisite state on which to assert. The regression suite could be reduced to one Cypress test file for each component. Any differences between this and the baseline snapshot would comprise a test failure.
```

Comment:
The regression suite could be reduced to one Cypress test file for each component. - it is already implemented [here](https://github.com/Sage/carbon/pull/2880)

Reply:
But we have multiple feature files for some components including `Button` and `Icon`? `Icon` in particular could be one single file called 'All Icons' like the 'All Buttons' one we have.